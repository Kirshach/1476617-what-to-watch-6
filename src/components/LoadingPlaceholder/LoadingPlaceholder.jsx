import React, {useEffect, useState} from 'react';

const getNewDots = (dots) => dots.length >= 3 ? `.` : dots + `.`;

const LoadingPlaceholder = () => {
  const [dots, setDots] = useState(`.`);
  useEffect(() => {
    const interval = setInterval(
        () => setDots((prevDots) => getNewDots(prevDots)), 1000
    );
    return () => clearInterval(interval);
  });

  return (
    <div className="loading-placeholder">
      Loading{dots}
    </div>
  );
};

export default LoadingPlaceholder
;
