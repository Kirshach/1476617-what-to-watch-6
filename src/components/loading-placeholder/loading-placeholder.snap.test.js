import React from 'react';
import {render} from '@testing-library/react';

import LoadingPlaceholder from './loading-placeholder';

test(`<LoadingPlaceholder /> renders correctly on initial mount`, () => {
  const {container} = render(<LoadingPlaceholder />);
  expect(container).toMatchSnapshot();
});
