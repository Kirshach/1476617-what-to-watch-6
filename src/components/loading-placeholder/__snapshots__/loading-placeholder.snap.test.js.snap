// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingPlaceholder /> renders correctly on initial mount 1`] = `
<div>
  <div
    class="loading-placeholder"
  >
    Loading
    .
  </div>
</div>
`;
